sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -778,6 +879,39 @@ class kernel_bundle_impl {
     return AdjustedName;
   }
 
+  bool ext_oneapi_has_device_global(const std::string &Name) {
+    std::string MangledName = mangle_device_global_name(Name);
```

Comment:
Do we currently check this part?
>  only true if kernel_bundle was created from source

Would that enable an early return, avoiding the linear search?

Reply:
A non-RTC bundle would just have an empty list of names, so the linear search isn't really a problem, but we can at least detect that before doing the string operations (and map lookups in case of kernels); see 1be7568.